src/lib/libcmd/cat.c

```diff
@@ -461,10 +463,9 @@ int b_cat(int argc, char **argv, Shbltin_t *context) {
         dovcat = 1;
     }
     if (flags & d_FLAG) sfopen(sfstdout, NULL, "wt");
-    cp = *argv;
-    if (cp) argv++;
-    do {
-        if (!cp || !strcmp(cp, "-")) {
+    if (!*argv) argv = stdin_argv;
```

Comment:
Why do we need this assignment ?

Reply:
Ok, I presume it's because we want to run this loop at least once.

Reply:
Precisely. I wouldn't necessarily write it this way in a different context. But given that `cat` by definition treats a single `-` argument as meaning "read from stdin" it's the simplest approach in this case. Although to be honest I'm inclined to do an even bigger refactoring because the body of this loop should be a separate function that is called for each `argv` element and is called once if argv is empty. But I felt it better to leave that refactoring for a subsequent change.